app/src/main/java/org/mozilla/fenix/gleanplum/DefaultMessageController.kt

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.gleanplum
+
+import android.content.Intent
+import androidx.annotation.VisibleForTesting
+import androidx.core.net.toUri
+import org.mozilla.fenix.BuildConfig
+import org.mozilla.fenix.HomeActivity
+
+/**
+ * Handles default interactions with an ui message.
+ */
+class DefaultMessageController(
+    private val messageManager: MessagesManager,
+    private val homeActivity: HomeActivity
+) : MessageController {
+
+    override fun onMessagePressed(message: Message) {
+        // TODO: report telemetry event
+        messageManager.onMessagePressed(message)
+        handleAction(message.data.action)
```

Comment:
@jhugman when I try to access `message.data.action` do I get the deeplink uri of key for the action? E.G
https://github.com/mozilla-mobile/fenix/pull/24048/files#diff-680637c1a9cdbef47ee7a3333ef917b741c19c3266663c0c760c0cbbc69b1928R129-R131

Reply:
See for more context 
https://github.com/mozilla-mobile/fenix/pull/24048/files#diff-337feac155df599668495c000b11628dbdf2d734512c3d6aece265a59a7a3a9bR39-R49

Reply:
What will happen for a web URL when accessing `message.data.action`?